code/modules/projectiles/projectile.dm

```diff
@@ -117,6 +121,18 @@
 		return 0
 	if(alwayslog)
 		add_attack_logs(firer, target, "Shot with a [type]")
+	if(knockback && ismovable(target))
+		if(!breaching && !isliving(target))
+			return
+		var/atom/movable/M = target
+		if(M.move_resist < INFINITY)
+			var/turf/start_turf = starting || get_turf(src)
+			var/throw_dir = get_dir(start_turf, get_turf(target))
+			var/atom/throw_target = get_edge_target_turf(get_turf(src), throw_dir)
+			M.throw_at(throw_target, knockbackdistance, knockbackspeed)
+		else
+			return
```

Comment:
```suggestion
		if(breaching && isliving(target))
		    var/atom/movable/M = target
		    if(M.move_resist < INFINITY)
			    var/turf/start_turf = starting || get_turf(src)
			    var/throw_dir = get_dir(start_turf, get_turf(target))
			    var/atom/throw_target = get_edge_target_turf(get_turf(src), throw_dir)
			    M.throw_at(throw_target, knockbackdistance, knockbackspeed)
```

Reply:
We don't want to return here, as it will stop the rest of the hit code from running

Reply:
the code is fixed now thanks to sean